pkg/agent/params.go

```diff
@@ -220,15 +221,11 @@ func assignKubernetesParameters(properties *datamodel.Properties, parametersMap
 
 		if servicePrincipalProfile != nil {
 			addValue(parametersMap, "servicePrincipalClientId", servicePrincipalProfile.ClientID)
-			keyVaultSecretRef := servicePrincipalProfile.KeyvaultSecretRef
-			if keyVaultSecretRef != nil {
-				addKeyvaultReference(parametersMap, "servicePrincipalClientSecret",
-					keyVaultSecretRef.VaultID,
-					keyVaultSecretRef.SecretName,
-					keyVaultSecretRef.SecretVersion)
-			} else {
-				addValue(parametersMap, "servicePrincipalClientSecret", servicePrincipalProfile.Secret)
-			}
+			encodedServicePrincipalClientSecret := base64.StdEncoding.EncodeToString([]byte(servicePrincipalProfile.Secret))
+			addValue(parametersMap, "servicePrincipalClientSecret", servicePrincipalProfile.Secret)
```

Comment:
`servicePrincipalClientSecret` is still used by Linux in several fields, and Linux use their ways to handle escape characters. @xuto2  @bowang-666 ,  do we also need encodedServicePrincipalClientSecret?